lib/vehicle_factory.rb

```diff
@@ -0,0 +1,89 @@
+
+class VehicleFactory
+  attr_accessor :most_popular_make_model,
+                :most_common_year,
+                :years,
+                :most_common_county
+               
+
+  def initialize
+    @most_popular_make_model = ""
+    @most_common_year = ""
+  end
+
+  def create_vehicles(input_api)
+    created_vehicles = []
+    if input_api.first[:state_of_residence] == "WA"
+      input_api.each do |hash|
+        hash_final = {}
+        hash_final[:vin] = hash[:vin_1_10]
+        hash_final[:year] = hash[:model_year]
+        hash_final[:make] = hash[:make]
+        hash_final[:model] = hash[:model]
+        hash_final[:engine] = :ev
+        created_vehicles << Vehicle.new(hash_final)
+      end
+    elsif input_api.first[:"state"] == "NY"
+      input_api.each do |hash|
+        if hash[:"record_type"] == "VEH"
+          hash_final = {}
+          hash_final[:vin] = hash[:vin]
+          hash_final[:year] = hash[:model_year]
+          hash_final[:make] = hash[:make]
+          hash_final[:model] = hash[:body_type]
+          hash_final[:engine] = hash[:fuel_type]
+          created_vehicles << Vehicle.new(hash_final)
+        end
+      end
+    end
+
+    # find the most popular make/model
+    all_makes_models = []
+    created_vehicles.each do |vehicle|
+      make_model = []
+      make_model << "#{vehicle.make}, #{vehicle.model}"
+      all_makes_models << make_model
+    end
+        
+    models_count = Hash.new(0)
+    all_makes_models.each do |model| 
+      models_count[model] += 1
+    end
+    @most_popular_make_model = (models_count.sort_by { |model,number| number}.last[0]).join
```

Comment:
Really fun to see this! You could pull this into a helper method also rather than having it be a part of your create method.